### PR TITLE
Add `Mockable` macro

### PR DESCRIPTION
**Pull request by pepicrft**
### Short description 📝
I'm adding the `Mockable` Swift Macro to improve the ergonomics of adding mocks. Having Swift Macros in the Tuist project is also useful to dog-food Tuist even further.

**Note** I still need to figure out why the build setting `SWIFT_ACTIVE_COMPILATION_CONDITIONS` is not being set.

### Contributor checklist ✅

- [x] The code has been linted using run `mise run lint:fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: Sources/TuistSupport/Utils/FileUnarchiver.swift [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import Foundation
 import TSCBasic
+import Mockable
 
-/// An interface to unarchive files from a zip file.
+@Mockable
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply Sources/TuistSupport/Utils/FileUnarchiver.swift:5]: Done :)

**Review by danyf90 — APPROVED** [review]

File: Project.swift [hunk 2]
```diff
@@ -135,6 +136,7 @@ func targets() -> [Target] {
                 .external(name: "ArgumentParser"),
                 .external(name: "GraphViz"),
                 .external(name: "AnyCodable"),
+                .external(name: "MockableTest"),
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply Project.swift:139]: No drawback. I think it's a good idea.

File: Package.swift [hunk 3]
```diff
@@ -265,6 +265,12 @@ var targets: [Target] = [
             "TSCclibc": .staticFramework,
             "TSCLibc": .staticFramework,
             "ArgumentParser": .staticFramework,
+            "Mockable": .staticFramework,
+            "MockableTest": .staticFramework,
+        ],
+        targetSettings: [
+            "Mockable": ["ENABLE_TESTING_SEARCH_PATHS": "YES"],
+            "MockableTest": ["ENABLE_TESTING_SEARCH_PATHS": "YES"],
         ],
```
Comment (danyf90) [inline Package.swift:274]: I would just add them to the "default" mapping [here](https://github.com/tuist/tuist/blob/a61e6e59a53c62e70fa9233d00cc4806c0f77045/Sources/TuistLoader/SwiftPackageManager/PackageInfoMapper.swift#L437)

**Review by danyf90 — APPROVED** [review]

**Review by danyf90 — APPROVED** [review]

File: Tests/TuistKitTests/Services/Plugin/PluginArchiveServiceTests.swift [hunk 4]
```diff
@@ -108,7 +108,9 @@ final class PluginArchiveServiceTests: TuistUnitTestCase {
         // Then
         XCTAssertEqual(invokedPackagePath, path)
         XCTAssertEqual(builtProducts, ["tuist-one", "tuist-two"])
-        XCTAssertEqual(fileArchiver.invokedZipParameters?.name, "TestPlugin.tuist-plugin.zip")
+
+        _ = verify(fileArchiver).zip(name: .value("TestPlugin.tuist-plugin.zip"))
```
Comment (Kolos65) [inline Tests/TuistKitTests/Services/Plugin/PluginArchiveServiceTests.swift:112]: @pepicrft 
This should have been:
```swift
verify(fileArchiver).zip(name: .value("TestPlugin.tuist-plugin.zip")).called(count: .atLeastOnce)
```
Without the last part there is no verification happening.

Comment (Kolos65) [inline Tests/TuistKitTests/Services/Plugin/PluginArchiveServiceTests.swift:112]: @pepicrft 
This should have been:
```swift
verify(fileArchiver).zip(name: .value("TestPlugin.tuist-plugin.zip")).called(count: .atLeastOnce)
```
Without the last part there is no verification happening.

**Pull request merged**